src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -401,27 +405,80 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitDataBlock(
       SparqlAutomaticParser::DataBlockContext* ctx) override {
-    return visitChildren(ctx);
+    // TODO: read through specs for correct behaviour of DataClause
+    if (ctx->inlineDataOneVar()) {
+      return GraphPatternOperation::Values{
+          std::move(visit(ctx->inlineDataOneVar()).as<SparqlValues>())};
+    } else if (ctx->inlineDataFull()) {
+      return GraphPatternOperation::Values{
+          std::move(visit(ctx->inlineDataFull()).as<SparqlValues>())};
+    }
+    AD_FAIL()  // Should be unreachable.
   }
 
   antlrcpp::Any visitInlineDataOneVar(
       SparqlAutomaticParser::InlineDataOneVarContext* ctx) override {
-    return visitChildren(ctx);
+    SparqlValues values;
+    auto var = visit(ctx->var()).as<Variable>();
+    values._variables.push_back(var.name());
+    for (auto& dataBlockValue : ctx->dataBlockValue()) {
+      values._values.push_back(
+          {std::move(visit(dataBlockValue).as<std::string>())});
+    }
+    return values;
   }
 
   antlrcpp::Any visitInlineDataFull(
       SparqlAutomaticParser::InlineDataFullContext* ctx) override {
-    return visitChildren(ctx);
+    SparqlValues values;
```

Comment:
I think you have to handle the NIL case where var is then a nullptr.
I think we can throw an exception there, because we don't support/understand these semantics at the moment.